aws/resource_aws_glue_catalog_database.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsGlueCatalogDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueCatalogDatabaseCreate,
+		Read:   resourceAwsGlueCatalogDatabaseRead,
+		Update: resourceAwsGlueCatalogDatabaseUpdate,
+		Delete: resourceAwsGlueCatalogDatabaseDelete,
+		Exists: resourceAwsGlueCatalogDatabaseExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"catalog_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"location_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"parameters": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueCatalogDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+	catalogID := createAwsGlueCatalogID(d, meta)
+	name := d.Get("name").(string)
+
+	input := &glue.CreateDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		DatabaseInput: &glue.DatabaseInput{
+			Name: aws.String(name),
+		},
+	}
+
+	_, err := glueconn.CreateDatabase(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Catalogue Database: %s", err)
```

Comment:
Nitpick: the service name is spelled `catalog` (American spelling).

Reply:
Ah noo! I'm an Australian living in London! I think the world would be a bit better off with a generous sprinkling of vowels here and there ;-P